service/src/test/java/bio/terra/pipelines/controller/PipelineRunsApiControllerTest.java

```diff
@@ -721,176 +723,249 @@ void getPipelineRunResultNotFound() throws Exception {
-    // wouldn't work and the test would fail
-    when(pipelineRunsServiceMock.findPipelineRunsPaginated(
-            100, requestPageToken, testUser.getSubjectId()))
-        .thenReturn(pageResponse);
-
-    MvcResult result =
-        mockMvc
-            .perform(
-                get(
-                    String.format(
-                        "/api/pipelineruns/v1/pipelineruns?limit=%s&pageToken=%s",
-                        limit, requestPageToken)))
-            .andExpect(status().isOk())
-            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
-            .andReturn();
-
-    ApiGetPipelineRunsResponse response =
-        new ObjectMapper()
-            .readValue(result.getResponse().getContentAsString(), ApiGetPipelineRunsResponse.class);
-
-    // response should one pipeline run
-    assertEquals(nextPageToken, response.getPageToken());
-    assertEquals(1, response.getResults().size());
-    ApiPipelineRun responsePipelineRun = response.getResults().get(0);
-    assertEquals(pipelineRun.getStatus().name(), responsePipelineRun.getStatus());
-    assertEquals(pipelineRun.getDescription(), responsePipelineRun.getDescription());
-    assertEquals(pipelineRun.getJobId(), responsePipelineRun.getJobId());
-  }
-
-  @Test
-  void getAllPipelineRunsWithTotalResults() throws Exception {
-    int limit = 10;
-    String pageToken = null;
-    PipelineRun pipelineRunPreparing1 = getPipelineRunPreparing(null);
-    PipelineRun pipelineRunPreparing2 = getPipelineRunPreparing(null);
-    PipelineRun pipelineRunPreparing3 = getPipelineRunPreparing(null);
-    PipelineRun pipelineRunPreparing4 = getPipelineRunPreparing(null);
-    PageResponse<List<PipelineRun>> pageResponse =
-        new PageResponse<>(
-            List.of(
-                pipelineRunPreparing1,
-                pipelineRunPreparing2,
-                pipelineRunPreparing3,
-                pipelineRunPreparing4),
-            null,
-            null);
-
-    when(pipelineRunsServiceMock.findPipelineRunsPaginated(
-            limit, pageToken, testUser.getSubjectId()))
-        .thenReturn(pageResponse);
-
-    when(pipelineRunsServiceMock.getPipelineRunCount(testUser.getSubjectId())).thenReturn(4L);
-
-    MvcResult result =
-        mockMvc
-            .perform(get(String.format("/api/pipelineruns/v1/pipelineruns?limit=%s", limit)))
-            .andExpect(status().isOk())
-            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
-            .andReturn();
-
-    ApiGetPipelineRunsResponse response =
-        new ObjectMapper()
-            .readValue(result.getResponse().getContentAsString(), ApiGetPipelineRunsResponse.class);
-
-    assertNull(response.getPageToken());
-    assertEquals(4, response.getResults().size());
-    assertEquals(4, response.getTotalResults());
+  @Nested
+  @Deprecated
+  @DisplayName("getAllPipelineRuns v1 tests")
+  class GetAllPipelineRunsV1Tests {
```

Comment:
what do y'all think of this way of organizing tests for deprecated endpoints?

i think nesting them all in a subclass makes it easier to know which tests we'll need to rip out once the CLI moves over to v2 and we're ready to remove v1 entirely. but i know there are a million different ways to accomplish this so i'm curious if anyone has any different ideas

Reply:
this makes a lot of sense to me!